app/src/main/java/de/fraunhofer/fokus/OpenMobileNetworkToolkit/Ping/PingParser.java

```diff
@@ -16,22 +19,37 @@ public class PingParser {
     private static final String TAG = "PingParser";
 
     private static PingParser instance = null;
-    private BufferedReader br;
+    private volatile BufferedReader br;
     private final ArrayList<PingInformation> lines;
 
     private final PropertyChangeSupport support;
     private PropertyChangeListener listener;
+
+    private static @Nullable WeakReference<PropertyChangeListener> externalPropertyChangeListener;
     private PingParser(BufferedReader br) {
         this.br = br;
         this.lines = new ArrayList<>();
         support = new PropertyChangeSupport(this);
+        if (externalPropertyChangeListener != null && externalPropertyChangeListener.get() != null) {
+            addPropertyChangeListener(externalPropertyChangeListener.get());
+        }
```

Comment:
while the comment is on point, we needed a way to set this thing *without* initializing the PingParser, or rather before it gets initialized by the PingWorker. the PingMonitor should set a listener to it (without having an instance) so that all future PingParsers can just call back into the PingMonitor